test/cypress/integration/widgets/blueprints_spec.js

```diff
@@ -5,145 +5,348 @@ describe('Blueprints widget', () => {
+            cy.contains('Show Data Types').click();
+            cy.contains('.modal button', 'Close').click();
+            const serverIp = '127.0.0.1';
+            cy.get('textarea').type(serverIp);
+            cy.contains('.modal .basic', 'Deploy').click();
+            cy.get('.modal').should('not.exist');
+
+            cy.wait('@deploy').then(({ requestBody }) => {
+                expect(requestBody).to.deep.equal({
+                    blueprint_id: emptyBlueprintName,
+                    inputs: { server_ip: serverIp },
+                    visibility: 'tenant',
+                    skip_plugins_validation: false,
+                    runtime_only_evaluation: false
+                });
+            });
+        });
 
-    it('should handle invalid blueprint url upload failure gracefully', () => {
-        cy.contains('Upload').click();
-        cy.get('input[name=blueprintUrl]').type('http://wp.pl').blur();
-        cy.contains('Cancel').click();
-    });
+        it('should allow to delete the blueprint', () => {
+            const blueprintName = `${blueprintNamePrefix}_delete`;
 
-    describe('should upload a blueprint', () => {
-        const url =
-            'https://github.com/cloudify-community/blueprint-examples/releases/download/5.0.5-65/utilities-examples-cloudify_secrets.zip';
+            cy.uploadBlueprint('blueprints/empty.zip', blueprintName);
+            getBlueprintRow(blueprintName).find('.trash').click();
+            cy.contains('.modal .button', 'Yes').click();
 
-        beforeEach(() => {
-            cy.contains('Upload').click();
-            cy.get('input[name=blueprintUrl]').type(url).blur();
+            cy.contains(blueprintName).should('not.exist');
         });
 
-        it('with default blueprint file', () => {
-            cy.get('input[name=blueprintUrl]').clear().blur();
-            cy.get('div[name=blueprintFileName] input').click();
-            cy.contains('No results found.');
-
-            cy.get('input[name=blueprintUrl]')
-                .type(
-                    'https://github.com/cloudify-community/blueprint-examples/releases/download/5.0.5-72/simple-hello-world-example.zip'
-                )
-                .blur();
-            cy.contains('.modal', 'blueprint.yaml');
-
-            cy.get('input[name=blueprintUrl]').clear().type(url).blur();
+        it('should do nothing on click when drill down is disabled', () => {
+            cy.editWidgetConfiguration('blueprints', () => cy.get('input[name=clickToDrillDown]').parent().click());
 
-            const blueprintName = `${blueprintNamePrefix}_default_file`;
-            cy.get('input[name=blueprintName]').clear().type(blueprintName);
-            cy.get('.button.ok').click();
+            cy.contains(emptyBlueprintName).click();
 
-            getBlueprintRow(blueprintName).contains('read-secret-blueprint.yaml');
+            cy.get('div.blueprintsWidget table.blueprintsTable');
```

Comment:
IMO the lone `get` looks line an unfinished line :smile: How about showing a bit more intent? Maybe I misunderstood something? :thinking: 

```suggestion
            cy.get('div.blueprintsWidget table.blueprintsTable').should('exist');
```

Reply:
The same applies to other bare `get`s without `should`

Reply:
As for me the intention is clear, after all we are using the same approach with `cy.contains`. Adding `should('exist')` seems a redundancy to me, but I have nothing against it after all - applied.